src/handlers/LP.ts

```diff
@@ -1,77 +1,77 @@
-import { AccountSnapshot } from "../schema/schema.ts";
+
+import { updatePoints } from "../points/point-manager.js";
+import { MISC_CONSTS, PENDLE_POOL_ADDRESSES } from "../consts.js";
+import { EthContext } from "@sentio/sdk/eth";
+import { LogLevel } from "@sentio/sdk";
+
+import { 
+  AccountSnapshotLP,
+  RateSnapshotLP,
+  RerunSnapshot,
+} from "../schema/schema.ts";
+
 import {
   PendleMarketContext,
   RedeemRewardsEvent,
   SwapEvent,
   TransferEvent,
   getPendleMarketContractOnContext,
 } from "../types/eth/pendlemarket.js";
-import { updatePoints } from "../points/point-manager.js";
+
 import {
   getUnixTimestamp,
   isLiquidLockerAddress,
   isSentioInternalError,
-  getAllAddresses,
+  getAllLPAddresses,
 } from "../helper.js";
-import { PENDLE_POOL_ADDRESSES } from "../consts.js";
-import { EthContext } from "@sentio/sdk/eth";
+
 import {
   readAllUserActiveBalances,
   readAllUserERC20Balances,
 } from "../multicall.js";
-import { EVENT_USER_SHARE, POINT_SOURCE_LP } from "../types.js";
 
-/**
- * @dev 1 LP = (X PT + Y SY) where X and Y are defined by market conditions
- * So same as Balancer LPT, we need to update all positions on every swap
- *
- * Users can further deposit LP to liquid lockers to get back receipt tokens.
- * This should also be handled here.
- *
- * Currently for all liquid lockers, 1 receipt token = 1 LP
- */
-
-export async function handleLPTransfer(
-  evt: TransferEvent,
-  ctx: PendleMarketContext
-) {
-  await processAllLPAccounts(ctx, [
-    evt.args.from.toLowerCase(),
-    evt.args.to.toLowerCase(),
-  ]);
-}
+import {
+  EVENT_USER_SHARE,
+  POINT_SOURCE_LP,
+  EVENT_POINT_INCREASE,
+  POINT_SOURCE,
+  POINT_SOURCE_YT,
+} from "../types.js";
 
-export async function handleMarketRedeemReward(
-  evt: RedeemRewardsEvent,
-  ctx: PendleMarketContext
-) {
-  await processAllLPAccounts(ctx);
-}
+const RATE_KEY = `RATES:${POINT_SOURCE_LP}`;
+const RERUN_KEY = `RERUN:${POINT_SOURCE_LP}`;
 
-export async function handleMarketSwap(_: SwapEvent, ctx: PendleMarketContext) {
-  await processAllLPAccounts(ctx);
-}
+/**
+ * @dev This function calculates the cumulative rate to convert LP into equivilent SY
+ * This function calculates three different rates:
+ * 1. the rate for liquid lockers - penpie
+ * 2. TODO: the rate for liquid lockers - EQB
+ * 3. TODO: the rate for the Zircuit points (time)
+ * and update the three different rates + timestamp to data store
+ */
+export async function updateLPtoSYRates(ctx: EthContext) {
```

Comment:
sorry, realized using `updateLPtoSYRates` would be confusing, since code viewers would be expecting calculation from SY to points, where in fact is included here
```suggestion
export async function updateLPtoPointRates(ctx: EthContext) {
```

Reply:
ok will update it